communication/src/actor.rs

```diff
@@ -377,8 +377,9 @@ mod test {
             {
                 self.has_received_response = true;
             } else if let CommunicationEvent::ConnectPeerResult { addr: _, result } = msg {
+                let peer_id = result.expect("Panic due to no network connection");
```

Comment:
Nice one. I assume this resolves the issue I faced with little-snitch blocking...

Reply:
The thing is that this test can only be successful if the process can use localhost, so if little-snitch is preventing that, it is correct that this should panic. But at least there is now a proper error message on that, if we want to handle this in a better way, I will have to either add a flag or some sort of delay to give little-snitch the time to allow it.